### DISCUSSION TITLE
Question : How to create a graphical console with Cosmos?

### DISCUSSION THREAD

**Discussion by raphm4044** [CosmosOS/Cosmos#2196] [Q&A]
#### Area of Cosmos - What area of Cosmos are we dealing with?
CGS

#### Actual Behaviour - What unexpectedly happens?
How to create a graphical console with Cosmos?

#### Version - Were you using the User Kit or Dev Kit? And what User Kit version or Dev Kit commit (Cosmos, IL2CPU, X#Last )?
Last User Kit

**Comment by MishaTy** [comment]
See this: https://github.com/MishaTY/Misha-OS/blob/master/Misha-OS/Gui/Apps/Terminal.cs

**Comment by raphm4044** [comment]
Misha, I don't want make a GUI OS. i want make the graphical console like AuraOS

**Comment by MishaTy** [comment]
You can use the same thing, just only drawing text

**Comment by raphm4044** [comment]
And , you can add a ScrollUp function to the Canvas class?
